dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -72,7 +82,13 @@ def __init__(
             raise ValueError("approximant should be a string, but got", approximant)
         else:
             self.approximant_str = approximant
-            self.approximant = LS.GetApproximantFromString(approximant)
+            self.lal_params = None
+            try:
+                self.approximant = LS.GetApproximantFromString(approximant)
+                if mode_list is not None:
+                    self.lal_params = self.setup_mode_array(mode_list)
+            except:
+                pass
```

Comment:
Why was this moved from where it was into the `else` and `try`/`except`?

Reply:
With non-LAL approximants, LS.GetApproximantFromString produces an error.

Reply:
And now SEOBNRv5PHM will pass through this code, since it initializes WaveformGenerator, so I though it was safer this way